### ISSUE TITLE
re-receiving notifications many times

### ISSUE THREAD

**Issue by anaderi** [ghaiklor/telegram-bot-github#75]
is it normal that I receive several notifications of the same thing? (i.e. new PR can be shown in _every_ message of the bot until I actually clear the notification) Can it be altered? I clearly want to be notified only once.

**Comment by ghaiklor** [comment]
@anaderi I'm checking only the `If-Modified-Since` header, so yes, may be that you are getting such kind of notifications.

https://github.com/ghaiklor/telegram-bot-github/blob/dev/src/services/GitHubNotifications.js#L39

**Comment by anaderi** [comment]
Thanks! are there any ideas on how to keep track of what has been sent?

**Comment by ghaiklor** [comment]
@anaderi hmm... what comes in first place, we can create in-memory object which will keeps that track of processed notifications. Not the best solution, but it easily can be done by keeping notification ids in an object\array and check for existence of id before sending a message.

**Comment by anaderi** [comment]
that would be awesome (although such cache may eat a lot of memory if many active github users subscribe to the bot notifications ... maybe you can keep latest 1000 or so?)

**Comment by ghaiklor** [comment]
@anaderi or... what if we will store latest timestamp into database.

Here I'm using date from response - https://github.com/ghaiklor/telegram-bot-github/blob/dev/src/services/GitHubNotifications.js#L39

```js
if (response.statusCode === 304 || response.statusCode === 200) {
  this._headers['If-Modified-Since'] = response.headers.date;
}
```

You can try to replace `If-Modified-Since` header with timestamp which will be stored in database as the latest timestamp you requested.

**Comment by anaderi** [comment]
would be even better )

**Comment by arieljannai** [comment]
Is it possible for a user to change that behavior?
If I'm not entering the notification in the near time, I can got tens of notifications on the same github notification.

**Comment by ghaiklor** [comment]
@arieljannai @anaderi I definitely need to find time for doing this, though, right now, I'm not able to investigate the issue. Feel free to make a PR and I'll deploy new version, thanks.

**Comment by arieljannai** [comment]
Ok, Hope I'll have time to look at this!
Thanks 😄

**Comment by slinkydeveloper** [comment]
Me too! I received same notification 2 times with 1 minute interval, next i received another notification 3 times with 1 hour interval

**Comment by sfarzy** [comment]
Ping, any update on this ?

**Comment by felixdivo** [comment]
The problem still exists.

**Comment by ghaiklor** [comment]
Should be fixed with #139
